Src/ILGPU/Runtime/Cuda/CudaAccelerator.cs

```diff
@@ -260,7 +260,7 @@ private void SetupAccelerator()
             CudaException.ThrowIfFailed(
                 CurrentAPI.GetDeviceName(out string name, DeviceId));
             Name = name;
-            DefaultStream = new CudaStream(this, IntPtr.Zero);
+            DefaultStream = new CudaStream(this, StreamFlags.CU_STREAM_NON_BLOCKING);
```

Comment:
ISSUE: The default Cuda stream is supposed to use Stream #0.
```suggestion
            DefaultStream = new CudaStream(this, IntPtr.Zero, responsible: false);
```

Reply:
shouldn't it be responsible of disposing itself?

Reply:
ILGPU does not actually create a default stream in Cuda. It is passing 0 as the stream, and Cuda interprets this as the default stream. Also, the existing dispose code checks for `streamPtr != IntPtr.Zero`, so it doesn't really mater what value we set for `responsible`.

Reply:
I see. Makes sense.